### PR TITLE
fix InferType for non pointer argument

### PR DESCRIPTION
**Pull request by rjodinchr**
clvk/api_tests is failing compiling the test added by this PR.

### CODE REVIEW

**Comment by rjodinchr** [comment]
@kpet  can you have a look at this PR please?

File: lib/Types.cpp [hunk 1]
```diff
@@ -317,8 +317,10 @@ Type *clspv::InferType(Value *v, LLVMContext &context,
           dyn_cast<MDString>(access_qual_op)->getString();
 
       if (access_qual_str == "none") {
-        assert(type_name_str.endswith("*") && "Only expect pointer types here");
-        auto type_name = type_name_str.drop_back(1);
+        auto type_name = type_name_str;
```
Comment (alan-baker) [inline lib/Types.cpp:320]: the first bit of InferType is:
```
if (!isa<PointerType>(v->getType()) {
  return v->getType();
}
```

So how is this getting hit?

Comment (rjodinchr) [reply lib/Types.cpp]: I think I found the real bug.

We were missing a break, which lead to overriding `ordinal`

**Review by alan-baker — APPROVED** [review]

**Review by kpet — COMMENTED** [review]
LGTM, thanks! That loop isn't the easiest to read. Hopefully that's the last of it.

**Pull request merged**